trpcultivate_phenotypes/src/Plugin/Validators/genusExists.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+/**
+ * @file
+ * Contains validator plugin definition.
+ */
+
+namespace Drupal\trpcultivate_phenotypes\Plugin\Validators;
+
+use Drupal\trpcultivate_phenotypes\TripalCultivateValidator\TripalCultivatePhenotypesValidatorBase;
+use Drupal\trpcultivate_phenotypes\Service\TripalCultivatePhenotypesGenusOntologyService;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Validate that genus exists and is configured.
+ *
+ * @TripalCultivatePhenotypesValidator(
+ *   id = "trpcultivate_phenotypes_validator_genus_exists",
+ *   validator_name = @Translation("Genus Exists and Configured Validator"),
+ *   input_types = {"metadata"}
+ * )
+ */
+class genusExists extends TripalCultivatePhenotypesValidatorBase implements ContainerFactoryPluginInterface {
+  /**
+   * Genus Ontology Service;
+   */
+  protected $service_PhenoGenusOntology;
+
+  /**
+   * Constructor.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition,
+    TripalCultivatePhenotypesGenusOntologyService $service_genus_ontology) {
+
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    
+    // Genus ontology service.
+    $this->service_PhenoGenusOntology = $service_genus_ontology;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition){
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('trpcultivate_phenotypes.genus_ontology')
+    );
+  }
+
+  /**
+   * Validate that genus provided exists and configured.
+   * 
+   * @param array $form_values
+   *   The values entered to any form field elements implemented by the importer.
+   *   Each form element value can be accessed using the field element key
+   *   ie. field name/key genus - $form_values['genus'].
+   * 
+   *   This array is the result of calling $form_state->getValues().
+   *
+   * @return array
+   *   An associative array with the following keys.
+   *     - case: a developer focused string describing the case checked.
+   *     - valid: either TRUE or FALSE depending on if the genus value is valid or not.
+   *     - failedItems: the failed genus value provided. This will be empty if the value was valid.
+   */
+  public function validateMetadata($form_values) {
+    // This genus validator assumes that a field with name/key genus was
+    // implemented in the Importer form.
+    $expected_field_key = 'genus';
+  
+    // Parameter passed to the method is not an array.
+    if (!is_array($form_values)) {
+      throw new \Exception(t('Unexpected @type type was passed as parameter to genusExists validator.', ['@type' => gettype($form_values)]));
```

Comment:
Remember not to use translation in the exception messages as they are meant for the developer. This needs to be fixed in a few other places in this PR as well.
```suggestion
      $type = gettype($form_values);
      throw new \Exception("Unexpected '$type' type was passed as parameter to genusExists validator when we expected an array.");
```